### DISCUSSION TITLE
Indirect dependency on privacy-violating SponsorLink package?

### DISCUSSION THREAD

**Discussion by egil** [dotnet/maui#16629] [General] [answered]
It looks like this project has an indirect dependency on the https://www.nuget.org/packages/Devlooped.SponsorLink/ package via the dependency on the https://www.nuget.org/packages/GitInfo package. 

I am not sure if SponsorLink's [questionable behavior](https://www.reddit.com/r/dotnet/comments/15ljdcc/does_moq_in_its_latest_version_extract_and_send/) impacts the users or devs of MAUI though, but probably something that should be investigated regardless.

https://github.com/dotnet/maui/blob/7c665392ab57159769617b3a404a57e73b546973/eng/Git.Build.targets#L3

**Comment by sjordanGSS** [comment]
Your "questionable behaviour" link doesn't seem to go anywhere

**Reply by egil** [reply]
Fixed. Here you can find more details: https://seankilleen.com/2023/08/on-moq-and-our-part-in-the-oss-sustainability-social-contract/

**Comment by jonpryor** [comment, marked as answer]
@egil wrote:

> It looks like this project has an indirect dependency on [SponsorLink]…

Does it? MAUI is explicitly using [GitInfo 2.2.0](https://github.com/devlooped/GitInfo/tree/v2.2.0/), which contains no occurrences of "SponsorLink".  SponsorLink wasn't added until https://github.com/devlooped/GitInfo/commit/5be5956bd044a0f13c33856190a3fc3efa6c4475 in GitInfo v3.0.0.  Versions if GitInfo before 3.0.0 do not use SponsorLink (unless the binary was updated after the fact, and I'm not sure that's possible).

Now, if MAUI were to update to GitInfo >= v3.0.0, *then* SponsorLink would be used, but this feels highly unlikely to happen at this point in time.

Additionally, MAUI *does not use `nuget.org`* to resolve packages: https://github.com/dotnet/maui/blob/1541190db7aa36ec560037f45cd169a64acbb4cd/NuGet.config#L16-L25

The packages sources MAUI uses do not provide GitInfo v3.0.0 or later.

**Reply by egil** [reply]
You are right. Didn't check the version numbers before posting. I suspect the author of GitInfo may remove the SponsorLink package in a later release due to the backlash.

**Reply by jfversluis** [reply]
In addition here is our response also on the .NET runtime repo which is also valid for this repository: https://github.com/dotnet/runtime/issues/90222#issuecomment-1671834283
